### ISSUE TITLE
avoid_catches_without_on_clauses should not warn if there is a rethrow

### ISSUE THREAD

**Issue by zoechi** [dart-lang/linter#909]
```dart
    try {
      incrementPendingRequestCounter();
      result = await serverRequest();
      decrementPendingRequestsCounter();
      return result;
      // ignore: avoid_catches_without_on_clauses
    } catch (e) {
      decrementPendingRequestsCounter(error: e.message);
      rethrow;
    }
```

**Comment by srawlins** [comment]
I'm all for this, and I think in particular it should not warn if there is a `rethrow` as one of the statements directly in the `catch` body. Meaning if a `rethrow` is found in an if-body, or whatever else, then `avoid_catches_without_on_clauses` is still reported. I don't want people is-checking the error inside an unguarded catch.
